### PR TITLE
NEX-25: Update to next-drupal 1.6.0, next 1.6.0 and next.js 13

### PR DESCRIPTION
**Pull request by vermario**
*Link to ticket: https://wunder.atlassian.net/browse/NEX-25*

*Changes proposed in this PR:*

* fix storybook setup, which was broken by the introduction of the next-i18next package and did not understand typescript module imports
* update the next-drupal package to latest (1.6.0) in next
* update to next.js 13 
    * use the new Link component (using codemod)
    * use the new Image component (using codemod)
* update to the next drupal module latest version
   *  updated exported configuration in the recipe with revalidator path, updated our custom hook
 

*How to test:*
* Follow the readme, create some content and see it displayed in the site.

### CODE REVIEW

File: next/.storybook/main.js [hunk 1]
```diff
@@ -1,15 +1,51 @@
...
-    "builder": "@storybook/builder-webpack5"
-  }
-}
\ No newline at end of file
+  framework: "@storybook/react",
+  core: {
+    builder: "@storybook/builder-webpack5",
+  },
+  webpackFinal: async (config) => {
+    config.resolve.plugins = [
```
Comment (vermario) [inline next/.storybook/main.js:31]: This config and the next are taken from https://theodorusclarence.com/blog/nextjs-storybook-tailwind

File: next/.storybook/main.js [hunk 2]
```diff
@@ -1,15 +1,51 @@
...
-  "core": {
-    "builder": "@storybook/builder-webpack5"
-  }
-}
\ No newline at end of file
+  framework: "@storybook/react",
+  core: {
+    builder: "@storybook/builder-webpack5",
+  },
+  webpackFinal: async (config) => {
```
Comment (vermario) [inline next/.storybook/main.js:30]: This one will make storybook recognize paths like "components/somecomponent", taken from https://github.com/storybookjs/storybook/issues/14087#issuecomment-1339108022

File: drupal/recipes/wunder_next_setup/config/next.next_site.frontend.yml [hunk 3]
```diff
@@ -6,3 +6,5 @@ label: 'Next.js frontend'
 base_url: 'http://localhost:3000'
 preview_url: 'http://localhost:3000/api/preview'
 preview_secret: null
+revalidate_url: 'http://localhost:3000/api/revalidate'
```
Comment (vermario) [inline drupal/recipes/wunder_next_setup/config/next.next_site.frontend.yml:9]: This configuration is for now empty, I have made a new story to actually activate this functionality: https://wunder.atlassian.net/browse/NEX-26

File: next/.storybook/main.js [hunk 4]
```diff
@@ -1,15 +1,51 @@
...
+    config.resolve.alias = {
+      ...config.resolve.alias,
+      "next-i18next": "react-i18next",
+    };
+    /**
+     * Fixes font import with /
+     * @see https://github.com/storybookjs/storybook/issues/12844#issuecomment-867544160
+     */
+    config.resolve.roots = [
+      path.resolve(__dirname, "../public"),
```
Comment (juhov) [inline next/.storybook/main.js:46]: FYI: No need to change anything, but Storybook has [staticDirs](https://storybook.js.org/docs/react/configure/images-and-assets#serving-static-files-via-storybook-configuration) config that could be used for this too.

**Review by juhov — APPROVED** [review]
Everything seems to work after these updates/fixes 👍

**Pull request merged**